Gems/Archive/Code/Source/Clients/ArchiveReader.h

```diff
@@ -0,0 +1,217 @@
+        ReadRawFileOutcome ReadRawFileIntoBuffer(AZStd::span<AZStd::byte> fileBuffer, AZ::u64 offset,
+            AZ::u64 fileSize,
+            const ArchiveReaderFileSettings& fileSettings);
+
+        //! Decompressed the content from the input buffer
+        //! @param decompressionResultSpan span to populated with decompressed results
+        //! @param compressedInputSpan compressed data span to decompress
+        //! @param fileSettings settings which indicate the max number of decompression task
+        //! to use for decompressing the file content.
+        //! This parameter also contains settings for selecting an offset within the decompressed
+        //! file to start reading, as well as a cap on the number of bytes to read from that start offset
+        //! @param blockOffsetIndex start index into the block line table vector which contains
+        //! the compressed sizes of each 2 MiB block of the file
+        //! @param extractFileResult Contains the compressed size, raw offset within the archive and uncompressed
+        //! size of the file needed for extracting
+        //! @return result outcome with a span containing a view of the decompressed file data
+        //! within the offset range specified by
+        //! [ArchiveReaderFileSettings::m_startOffset,ArchiveReaderFileSettings::m_startOffset + ArchiveReaderFileSettings::m_bytesToRead)
+        //! Otherwise an error message string providing reasons why decompression failed
+        using ReadCompressedFileOutcome = AZStd::expected<AZStd::span<AZStd::byte>, ResultString>;
+        ReadCompressedFileOutcome ReadCompressedFileIntoBuffer(AZStd::span<AZStd::byte> decompressionResultSpan,
+            const ArchiveReaderFileSettings& fileSettings,
+            const ArchiveExtractFileResult& extractFileResult);
+
+
+        // Private Member variables section
+
+        //! Archive Reader specific settings
+        //! Controls the number of tasks to use for reading and decompression of content
+        //! from the archive
+        //! Also contains an error callback that is invoked when error occurs in the constructor
+        ArchiveReaderSettings m_settings;
+        //! Archive header as read from the first sizeof(ArchiveHeader) blocks of the archive stream
+        //! The header is not modified by the reader
+        ArchiveHeader m_archiveHeader;
+        //! View of the Archive TOC within the supplied archive stream
+        //! Since the ArchiveReader doesn't mutate the archive, a Table of Contents View is used
+        //! and paired with a raw buffer of the Table of Contents
+        struct ArchiveTableOfContentsReader
+        {
+            // Default a table of contents reader that has an empty vector
+            // and default constructed view
+            ArchiveTableOfContentsReader();
+            // Takes ownership of the buffer containing the Table of Contents raw data
+            // and an ArchiveTableOfContentsView instance
```

Comment:
'Takes ownership' is mentioned here as well

Reply:
I will update this comment as well to remove the mention of taking over ownership of the buffer